wire.go

```diff
@@ -15,7 +15,7 @@ import (
 // Typ3 and Typ4
 
 type Typ3 uint8
-type Typ4 uint8 // Typ3 | 0x80 (pointer bit)
+type Typ4 uint8 // Typ3 | 0x08 (pointer bit)
```

Comment:
This perhaps warrants us removing a bunch of these magic constants in favour of variables/consts
```go
const typ4Pointer uint8 = 0x08
```
and I like the new introduction of the method `IsPointer() bool`

Reply:
Yeah we should move over to methods everywhere.